### ISSUE TITLE
What options exist to optimize playing 4 mp4 files hosted on a pc over a network

### ISSUE THREAD

**Issue by manity42** [unosquare/ffmediaelement#418]
# Random Crash when playing multiple network streams

*Are there any properties or options to adjust on the control to help improve the play back of mp4 videos stored on a network hdd location , with the player run on a different PC, so playing mp4 files over a network and not streaming. The network can be laggy sometimes and im trying to open 4 videos at the same time, then move onto another 4 videos then move onto another 4 videos etc etc using 4 instances of the control in one application.
any properties i can try out.
Any advice on how to optimize loading many random groups of 4 x videos simultaneously and playing in this environment. *

im using 
FFME v 4.1.300
FFmpeg.Autogen 4.1.0.5
FFMPEG v4.13

Note  i did try version 4.1.310 but changed back.
While version 4.1.300 is generally reliable playing in the manner i described above over a network the new version 4.1.310 would just lock and  freeze then just disappear after the 4th group of videos loaded without any exceptions. Its the same code for both versions. And it very reliably crashes on the 4th group of videos loaded. and the video groups are randomly selected so as to rule out the media being the issue. however thats an issue for another time.
right now im just looking to improve stability and performance over a network that may be busy in the manner described above.

Any advice appreciated. 

## Issue Categories
 - [x ] Question

## Version Information

 - [X] NuGet Package *4.1.300*

**Comment by mariodivece** [comment]
On the settings for network playback, have you seen the discussion on #391 

On the crashing, I will investigate. It's a strange issue and I believe a serious one. I will try to reproduce.

**Comment by manity42** [comment]
thanks for the reminder of #391. i had forgotten it. I will review it.
On your advice do you think it would be better to, for the purpose of opening 4 videos at the same time,

1. use the await player.open(uri) option
2 use player.open(uri) option without the await
3. change the player.source = new URI option.
4 use a mutex blocking variable to allow only one player at a time to open a video at a time, keeping the others in a holding state and only after the media ready event is fired for a video opened, then allow the next player to open its media and so on. 
In this way only one player at a time would open media, the others would wait until the previous media has finished loading.

Of these options is one better than the others for opening 4 videos at the same time in your opinion.

thanks

**Comment by mariodivece** [comment]
Always prefer the `await Media.Open(url)` method.
In your place I would do something like this:

```csharp

private readonly List<MediaElement> Media = new List<MediaElement>(4);

public MyClass()
{
    Media.Add(Media1);
    Media.Add(Media2);
    Media.Add(Media3);
    Media.Add(Media4);
}

public async Task OpenVideosAsync(params string[] urls)
{
    for (var i = 0; i < urls.Length; i++)
    {
        if (i > Media.Count - 1) break;
        await Media[i].Open(urls[i]);
    }
}
```

And call `await OpenVideoAsync` from the UI thread.

**Comment by manity42** [comment]
I will close this issue. the problem i experienced of freezing and locking after updating to version 4.1.310 i resolved by creating a new project based on 4.1.310, and moving my code into the new project. I don't know why it was occurring after updating my old project by nuget. It maybe  my project files were corrupt in some way. 

I followed your advice in opening media in the work flow you suggested and that seems to have improved when opening/playing across a network.

for any one who is considering using the cache option as in issue #391, note that the nightly build of the FFMPEG lib after v4.13 has fixed the windows problem with the cache and the temp files it creates. When you activate the cache option in FFME a local cache file is created for each video opened, even when you open more than one at the same time. Those temp cache files are now auto deleted by FFMPEG libs when you close a video. Of course if the app crashes the temp files are not auto deleted, so you need to take of that.
